astropy/units/quantity.py

```diff
@@ -1687,7 +1687,8 @@ def _to_own_unit(self, value, check_precision=True, *, unit=None):
 
         # Setting names to ensure things like equality work (note that
         # above will have failed already if units did not match).
-        if self.dtype.names:
+        # TODO: is this the best place to do this?
+        if _value.dtype.names:
```

Comment:
Can this be the more explicit 
```suggestion
        if _value.dtype.names is not None:
```
?

Reply:
Yes, that is better. I also realized while continuing to work on the structured distribution that just setting the names is bad - it propagates backwards since the dtypes are shared by reference. So, I changed that (and added a test)